python/cudf/cudf/core/column/datetime.py

```diff
@@ -540,65 +526,56 @@ def isocalendar(self) -> dict[str, ColumnBase]:
             )
         }
 
-    def normalize_binop_value(  # type: ignore[override]
-        self, other: DatetimeLikeScalar
-    ) -> cudf.Scalar | cudf.DateOffset | ColumnBase:
-        if isinstance(other, (cudf.Scalar, ColumnBase, cudf.DateOffset)):
+    def _normalize_binop_operand(self, other: Any) -> pa.Scalar | ColumnBase:
+        if isinstance(other, (ColumnBase, cudf.DateOffset)):
             return other
-
-        tz_error_msg = (
-            "Cannot perform binary operation on timezone-naive columns"
-            " and timezone-aware timestamps."
-        )
-        if isinstance(other, pd.Timestamp):
-            if other.tz is not None:
-                raise NotImplementedError(tz_error_msg)
-            other = other.to_datetime64()
-        elif isinstance(other, pd.Timedelta):
-            other = other.to_timedelta64()
-        elif isinstance(other, datetime.datetime):
-            if other.tzinfo is not None:
-                raise NotImplementedError(tz_error_msg)
-            other = np.datetime64(other)
-        elif isinstance(other, datetime.timedelta):
-            other = np.timedelta64(other)
-
-        if isinstance(other, np.datetime64):
-            other_time_unit = np.datetime_data(other.dtype)[0]
-            if np.isnat(other):
-                if other_time_unit not in {"s", "ms", "ns", "us"}:
-                    other_time_unit = "ns"
-
-                return cudf.Scalar(
-                    None, dtype=f"datetime64[{other_time_unit}]"
-                )
-            if other_time_unit not in {"s", "ms", "ns", "us"}:
-                other = other.astype(self.dtype)
-
-            return cudf.Scalar(other)
-        elif isinstance(other, np.timedelta64):
-            other_time_unit = np.datetime_data(other.dtype)[0]
-            if np.isnat(other):
-                return cudf.Scalar(
-                    None,
-                    dtype="timedelta64[ns]"
-                    if other_time_unit not in {"s", "ms", "ns", "us"}
-                    else other.dtype,
-                )
-
-            if other_time_unit not in {"s", "ms", "ns", "us"}:
-                other = other.astype(
-                    np.dtype(f"timedelta64[{self.time_unit}]")
-                )
-
-            return cudf.Scalar(other)
-        elif isinstance(other, str):
-            try:
-                return cudf.Scalar(other, dtype=self.dtype)
-            except ValueError:
-                pass
-
-        return NotImplemented
+        elif is_scalar(other) or (
+            isinstance(other, (cp.ndarray, np.ndarray)) and other.ndim == 0
```

Comment:
You could avoid needing to do this check twice if you put all of this into the else and had it return NotImplemented at the very end instead of using an elif with a separate else clause.